src/io.c

```diff
@@ -759,7 +760,11 @@ int apk_dir_foreach_file(int dirfd, apk_dir_file_cb cb, void *ctx)
 			    (de->d_name[1] == '.' && de->d_name[2] == 0))
 				continue;
 		}
-		cb(ctx, dirfd, de->d_name);
+		ret = cb(ctx, dirfd, de->d_name);
+		if (ret != 0) {
```

Comment:
a bit worried that this could have some unforeseen effects, the unit tests pass however.

Was this what you meant or did I misunderstand?